File: zero-sum-subarray/ZeroSumSubArray.cs
using System;

public class ZeroSumSubArray
{
    public static bool ZeroSumSubArray(int[] array) {
        if(array.Length == 1 && array[0] == 0) { return true; }
        int index = 1;
        while(index < array.Length) {
            int leftIndex = index - 1;
            int sum = 0;
            while(leftIndex >= 0) {
                int leftSum = array[leftIndex] + array[index];
                if(leftSum == 0) { return true; }
                sum += leftSum;
                leftIndex--;
            }

            int rightIndex = index + 1;
            while(rightIndex < array.Length) {
                int rightSum = array[rightIndex] + array[index];
                if(rightSum == 0) { return true; }
                sum += rightSum;
                rightIndex++;
            }
            
            if(sum == 0) { return true; }

            index++;
        }
        return false;
    }
}

File: zero-sum-subarray/ZeroSumSubArray.swift
import Foundation

class ZeroSumSubArray {
    func zeroSumSubArray1(_ array: [Int]) -> Bool {
        if(array.count == 1 && array.first! == 0) { return true }
        for index in stride(from: 1, to: array.count, by: 1) {
            var sum = 0
            var leftIndex = index - 1;
            while(leftIndex >= 0) {
                var leftSum = array[leftIndex] + array[index]
                if(leftSum == 0) { return true }
                sum += leftSum
                leftIndex -= 1
            }

            var rightIndex = index + 1;
            while(rightIndex < array.count) {
                var rightSum = array[rightIndex] + array[index]
                if(rightSum == 0) { return true }
                sum += rightSum
                rightIndex += 1
            }

            if(sum == 0) { return true }
        }
        return false
    }

    func ZeroSumSubArray2(_ array: [Int]) -> Bool {
        var set: Set<Int> = []
        var summation = 0
        for num in array {
            summation += num
            if(num == 0 || summation == 0 || set.contains(summation)) { return true }
            set.insert(summation)
        }
        return false
    }
}